src/proxy/routes/chat_completions.py

```diff
@@ -70,35 +75,78 @@ def __include_router(self):
-        ) -> Any:
+        ) -> openai.openai_object.OpenAIObject | str | StreamingResponse:
             """OpenAI chat completion response"""
 
-            # see if extensions is in the path
-            if "/extensions/" in request.url.path:
-                chat.extensions = True
+            if model.api_version is None:
+                model.api_version = OPENAI_CHAT_COMPLETIONS_API_VERSION
 
-            # get the api version from the query string
-            if "api-version" in request.query_params:
-                chat.api_version = request.query_params["api-version"]
+            completion, status_code = await self.process_request(
+                deployment_id=deployment_id,
+                request=request,
+                model=model,
+                call_method=self.call_openai,
+                validate_method=self.__validate_chat_completion_request,
+            )
 
-            authorize_response = await self.authorize_request(deployment_id=deployment_id, request=request)
+            if isinstance(completion, AsyncGenerator):
+                return StreamingResponse(completion)
 
-            if chat.max_tokens and chat.max_tokens > authorize_response.max_token_cap:
-                chat.max_tokens = authorize_response.max_token_cap
+            response.status_code = status_code
+            return completion
 
-            (
-                completion,
-                status_code,
-            ) = await self.request_class_mgr.call_openai_chat_completion(chat)
+        return self.router
 
-            if isinstance(completion, AsyncGenerator):
-                return StreamingResponse(completion)
-            else:
-                response.status_code = status_code
-                return completion
+    async def call_openai(
+        self,
+        model: ChatCompletionsRequest,
+        openai_request: dict[str, Any],
+        deployment: Deployment,
+    ) -> tuple[openai.openai_object.OpenAIObject, int] | AsyncGenerator:
+        """call openai with retry"""
+
+        url = (
+            f"https://{deployment.resource_name}.openai.azure.com/openai/deployments/"
+            f"{deployment.deployment_name}/chat/completions"
+            f"?api-version={model.api_version}"
+        )
+
+        async_mgr = OpenAIAsyncManager(deployment)
+
+        if model.stream:
+            response, http_status_code = await async_mgr.async_post_streaming(openai_request, url)
+        else:
+            response, http_status_code = await async_mgr.async_openai_post(openai_request, url)
+            response["model"] = deployment.friendly_name
+
+        return response, http_status_code
+
+    def __validate_chat_completion_request(self, model: ChatCompletionsRequest):
+        """validate input"""
+
+        # check the max_tokens is between 1 and 4000
+        if model.max_tokens is not None and not 1 <= model.max_tokens <= 4000:
+            self.report_exception("Oops, max_tokens must be between 1 and 4000.", 400)
```

Comment:
I'm assuming this is a hold over from the current implementation, but we should revise that as the 16k and above models invalidate this.